src/controllers/collection.controller.ts

```diff
@@ -0,0 +1,28 @@
+import { Request,Response } from "express";
+
+import { createCollectionAttributes } from "../types/collection.types";
+import { CreateCollectionService } from "../services/collection.services";
+import { UserService } from "../services/user.services";
+import "../utils/cloudinary.utils"
+
+export const createCollection=async function(req:Request,res:Response){
+    const seller=await UserService.getUserByid(req.params.seller_id)
+
+    if(seller?.role!=='seller'){
+        return res.status(400).json({message:"You have to be a seller to create a collection"})
+    }
+
+    const collection:createCollectionAttributes={
+        CollectionName:req.body.CollectionName,
+        description:req.body.description,
+        seller_id:req.params.seller_id
```

Comment:
Ensure that the seller ID is retrieved from the currently authenticated user using req.user.id.